readme.md

```diff
@@ -295,13 +295,64 @@ Returns `true` if `value` is an even integer.
 
 Returns `true` if `value` is an odd integer.
 
-##### .empty(value)
+##### .emptyArray(value)
 
-Returns `true` if `value` is falsy or an empty string, array, object, map, or set.
+Returns `true` if the value is an array and the `length` is 0.
 
-##### .emptyOrWhitespace(value)
+##### .nonEmptyArray(value)
 
-Returns `true` if `is.empty(value)` or a string that is all whitespace.
+Returns `true` if the value is an array and the `length` is *not* 0.
+
+##### .emptyString(value)
+
+Returns `true` if the value is an empty string. `false` otherwise.
+
+##### .nonEmptyString(value)
+
+Returns `true` if the value is string, and the length !== 0. `false` when the string is empty.
+
+##### .emptyObject(value)
+
+Returns `true` if the value is an object and `Object.keys(value)` length is 0. 
+
+Please note that `Object.keys` returns only enumerable properties. Hence something like this can happen - 
+
+```js
+const object1 = {};
+
+Object.defineProperty(a, 'property1', {
```

Comment:
What's the purpose of the defineProperty in this example

Reply:
It is because `isEmptyObject` checks for only `enumerable` properties. Hence,  I added an example to show that an object can have non-enumerable properties and still the result will be empty.

Reply:
So change it so that defineProperty will run on `object1` and not on `a`